scripts/lib/docbook_template.py

```diff
@@ -0,0 +1,185 @@
+      <link xlink:href="{ITEM_URL}"></link>
+    </bibliosource>
+    <pubdate status="draft">{PUBLICATION_DATE}</pubdate>
+  </info>
+  {ITEM_TAG_LIST}
+  {ITEM_CONTENT}
+  {INCLUDE_ITEMS}
+</chapter>
+''',
+    section='''<section xml:id="section_{ITEM_ID}"
+      xmlns="http://docbook.org/ns/docbook"
+      version="5.0"
+      xml:lang="en"
+      xmlns:xlink="http://www.w3.org/1999/xlink"
+      xmlns:xi="http://www.w3.org/2001/XInclude"
+      xmlns:xl="http://www.w3.org/1999/xlink">
+  <info>
+    <title>{ITEM_TITLE}</title>
+    <bibliosource class="uri">
+      <link xlink:href="{ITEM_URL}"></link>
+    </bibliosource>
+    <pubdate status="draft">{PUBLICATION_DATE}</pubdate>
+  </info>
+  {ITEM_TAG_LIST}
+  {ITEM_CONTENT}
+  {INCLUDE_ITEMS}
+</section>
+''',
+    item_tag='<indexterm><primary>{TAG_NAME}</primary></indexterm>',
+)
+
+def level_to_part_type(level: int) -> str:
+    part_types = ["book", "chapter", "section"]
+
+    return part_types[level] if level < len(part_types) else part_types[-1]
+
+def get_date(iso_dt: str) -> str:
+    if iso_dt is None:
+        return ""
+
+    iso_dt = iso_dt.replace('Z', '+00:00')
+    date = datetime.fromisoformat(iso_dt)
+    return date.astimezone().strftime('%Y-%m-%d')
+    
+
+def get_person(full_name: str) -> list:
+    person = full_name.split()
+    
+    if len(person) == 1:
+        person.append("")
+        
+    return TEMPLATES["person"].format(
+        PERSON_NAME=person[0],
+        PERSON_SURNAME=person[1],
+    )
+    
+def path_escape(text: str) -> str:
+    return parser.quote(text)
+
+
+def xml_escape(text: str ) -> str:
+    return text.replace('&', "&amp;")
+
+
+def wrap_book(meta: dict, content: str, child_items: list[Path]) -> str:
+    book = dict(
+        ITEM_ID=meta["id"],
+        ITEM_TITLE=xml_escape(meta["title"]),
+        ITEM_URL=meta["url"],
+        PUBLICATION_DATE=get_date(meta["modified_at"]),
+        AUTHOR_DESC=get_person(meta["author"]),
+        ITEM_CONTENT=content,
+    )
+
+    coauthors = [
+        TEMPLATES["coauthor"].format(COAUTHOR_DESC=get_person(a))
+            for a in meta["coauthors"]
+    ]
+    book["COAUTHORS"] = "\n".join(coauthors)
+     
+    reviewers = [
+        TEMPLATES["reviewer"].format(REVIEWER_DESC=get_person(r))
+            for r in meta["reviewers"]
+    ]
+    book["REVIEWERS"] = "\n".join(reviewers)    
+    
+    child_items_xml = [
+        TEMPLATES["incl_item"].format(ITEM_PATH=path_escape(str(item_p)))
+            for item_p in child_items
+    ]
+    book["INCLUDE_ITEMS"] = "\n".join(child_items_xml)
+
+    return TEMPLATES["book"].format(**book)
+
+
+def wrap_part(
+        part_type: str,
+        meta: dict,
+        content: str, 
+        child_items: list,
+    ) -> str:
+
+    part = dict(
+        ITEM_ID=meta["id"],
+        ITEM_TITLE=xml_escape(meta["title"]),
+        ITEM_URL=meta["url"],
+        PUBLICATION_DATE=get_date(meta["modified_at"]),
+        ITEM_CONTENT=content,     
+    )
```

Comment:
Nit: good candidate for [dataclass](https://docs.python.org/3/library/dataclasses.html), for cases when we reading from json [pydantic](https://docs.pydantic.dev/latest/#pydantic-examples) can be used